lib/wallet/src/Cardano/Wallet.hs

```diff
@@ -2865,22 +2865,60 @@ delegationFee db@DBLayer{atomically, walletsDB} netLayer
                 availableUTxO @s mempty wallet
         pureTimeInterpreter <- lift $ snapshot ti
         stdGen <- initStdGen
-        feePercentiles <- calculateFeePercentiles $ do
-            (Cardano.Tx (Cardano.TxBody bodyContent) _, _updatedWallet) <-
-                buildTransactionPure @s @k @n @era
```

Comment:
(I inlined `buildTransactionPure` to be able to put the partial tx construction outside the loop)